HP/src/hp_main.f90

```diff
@@ -141,8 +133,18 @@ PROGRAM hp_main
         !
         ! If necessary the bands are recalculated
         !
+#if defined(__SIRIUS)
+        IF ( .not. setup_pw ) THEN
```

Comment:
What is the logic of `.not. setup_pw` here?

Reply:
To do nscf I am clearing up and resetting sirius at each iteration (mainly for the kset_handler, k+q is changing and I need to know it for nscf). This is set in PW/src/non_scf.f90 if I need to do the nscf, if not this part is skipped and I need to force the setup_sirius here with this if. I built up this logic so that we can use what is inside non_scf.f90 also using pw with calculation='nscf'. 

If I don't put this line there, we will face the breaking for the second perturbed atom. 
I don't know if it is worth to factorize a part of setup_sirius to do this, in principle what I need to do is just clean up kset_handler and reinitialize it, not the whole sirius interface.